plugins/woocommerce-admin/client/core-profiler/use-loader-stages.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import LightBulb from './assets/images/loader-lightbulb.svg';
+import { Stages } from './pages/Loader';
+
+export const useStages = ( key: string ): Stages => {
```

Comment:
I think we shouldn't name this with a `use` prefix because it's not actually a React hook. There are some lint rules that apply conditions based on the `use` prefix such as [these](https://legacy.reactjs.org/docs/hooks-rules.html), and they would be mistakenly applied there.

Perhaps we could call it `getLoaderStageMeta` ?

Reply:
Good point! I blindly copied the same pattern from processing-step component from wp-calypso 🤦 

Updated in https://github.com/woocommerce/woocommerce/pull/38019/commits/1b95e171f1f9a4e98c4979c5eca4f5f029b46a03